webapp/store/reducers/nav.js

```diff
@@ -10,14 +11,24 @@ const navStore = (state = initialState, action) => {
 
   switch (type) {
     case ADD_SIDE_NAV: {
-      const duplicates = newState.sidebar.some(
+      const sidebar = [...newState.sidebar];
+      // check if there are any duplicates. Will return true
+      // if at least one item in the sidebar has the same name or id
+      // as the item attempting to be added
+      const duplicates = sidebar.some(
         nav => nav.name === payload.name || (!!nav.id && nav.id === payload.id)
       );
 
       if (duplicates)
         // eslint-disable-next-line no-console
         console.error(`Can't have duplicate names or ids for nav items`);
-      else newState.sidebar = [...newState.sidebar, payload];
+      else {
+        const paths = sidebar.map(item => item.pathName);
```

Comment:
could do something like:

```
const { paths, names } = sidebar.reduce((item,acc) => {
  acc.paths.push(item.path)  // could also go more functional
  acc.names.push(item.name)
  return acc;
}, { paths: [], names: [] })
```

to save some iterations, but I don't think that these lists will be very large, so it shouldn't be that large of a deal